samples/Azure.Management.Storage/Generated/BlobContainerContainer.cs

```diff
@@ -5,23 +5,357 @@
 
 #nullable disable
 
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Management.Storage.Models;
 using Azure.ResourceManager.Core;
+using Azure.ResourceManager.Core.Resources;
 
 namespace Azure.Management.Storage
 {
     /// <summary> A class representing collection of BlobContainer and their operations over a StorageAccount. </summary>
-    public partial class BlobContainerContainer
+    public partial class BlobContainerContainer : ResourceContainerBase<TenantResourceIdentifier, BlobContainer, BlobContainerData>
     {
-        /// <summary> Initializes a new instance of BlobContainerContainer for mocking. </summary>
+        /// <summary> Initializes a new instance of the <see cref="BlobContainerContainer"/> class for mocking. </summary>
         protected BlobContainerContainer()
         {
         }
 
-        internal BlobContainerContainer(ResourceOperationsBase parent)
+        /// <summary> Initializes a new instance of BlobContainerContainer class. </summary>
+        /// <param name="parent"> The resource representing the parent resource. </param>
+        internal BlobContainerContainer(ResourceOperationsBase parent) : base(parent)
         {
+            _clientDiagnostics = new ClientDiagnostics(ClientOptions);
+            _pipeline = ManagementPipelineBuilder.Build(Credential, BaseUri, ClientOptions);
         }
 
+        private readonly ClientDiagnostics _clientDiagnostics;
+        private readonly HttpPipeline _pipeline;
+
+        /// <summary> Represents the REST operations. </summary>
+        private BlobContainersRestOperations _restClient => new BlobContainersRestOperations(_clientDiagnostics, _pipeline, Id.SubscriptionId);
+
+        /// <summary> Typed Resource Identifier for the container. </summary>
+        public new ResourceGroupResourceIdentifier Id => base.Id as ResourceGroupResourceIdentifier;
```

Comment:
This new shouldn't be needed

Reply:
Without the new, Id is just a `ResourceIdentifier` inherited from `OperationBase`. We need to cast it either here or at the level of `ContainerBase`

Reply:
A PR to arm.core